packages/react-ui-ag/src/Listings/DesktopMapPinListing.js

```diff
@@ -0,0 +1,155 @@
+  className: PropTypes.string,
+  valueLocation: PropTypes.string,
+})
+
+@themed(/^DesktopMapPinListing/, { pure: true })
+export default class DesktopMapPinListing extends PureComponent {
+  static propTypes = {
+    index: PropTypes.number,
+    listing: PropTypes.object,
+    theme: PropTypes.object,
+    className: PropTypes.string,
+    onClick: PropTypes.func,
+    photo: PropTypes.object,
+    propertyName: PropTypes.object,
+    ctaButtons: PropTypes.arrayOf(buttonPropTypes),
+    favoriteButton: buttonPropTypes,
+  }
+
+  static defaultProps = {
+    theme: {},
+    listing: {},
+    ratings: {},
+    photos: {},
+  }
+
+  @autobind
+  handleCardClick() {
+    const { index, listing, onClick } = this.props
+
+    if (onClick) onClick(index, listing)
+  }
+
+  @autobind
+  handleButtonClick(onClick) {
+    return event => {
+      if (onClick) onClick(this.props.listing)
+      if (event && event.stopPropagation) event.stopPropagation()
+    }
+  }
+
+  @autobind
+  handleFavoriteClick(value, event) {
+    const { favoriteButton: { onClick } } = this.props
+
+    if (onClick) onClick(this.props.listing, value)
+    if (event && event.stopPropagation) event.stopPropagation()
+  }
+
+  renderCtaButtons() {
+    const { ctaButtons } = this.props
+    return ctaButtons.map((cta, index) => this.renderCtaButton(cta, index))
+  }
+
+  renderCtaButton(props, key) {
+    const { theme, listing } = this.props
+    const {
+      className,
+      onClick,
+      valueLocation,
+      children,
+      ...rest
+    } = props
+
+    const buttonText = get(listing, valueLocation, children)
+    const buttonProps = { ...rest }
+
+    if (!buttonText) return null
+
+    return (
+      <Button
+        {...buttonProps}
+        className={classnames(
+          theme.DesktopMapPinListing_CtaButton,
+          className,
+        )}
+        onClick={this.handleButtonClick(onClick)}
+        key={key}
+        data-tid="cta-button"
+      >
+        {buttonText}
+      </Button>
+    )
+  }
+
+  renderFavoriteButton() {
+    const { theme, favoriteButton, listing } = this.props
+    const { className } = favoriteButton
+    return (
+      <ToggleButton
+        {...favoriteButton}
+        value={listing.isFavorited}
+        className={classnames(
+          theme.DesktopMapPinListing_FavoriteButton,
+          className,
+        )}
+        onClick={this.handleFavoriteClick}
+      />
+    )
+  }
+
+  render() {
+    const {
+      theme,
+      listing,
+      onClick,
+      className,
+      photo,
+      ctaButtons,
+      favoriteButton,
+      propertyName,
+      ...props
+    } = this.props
+
+    return (
+      <ListingCell
+        listing={listing}
+        onClick={this.handleCardClick}
+        className={classnames(
+          className,
+          theme.DesktopMapPinListing,
+        )}
+        {...props}
+      >
+        <ListingComponents.Photo {...photo} />
+        <div className={theme.DesktopMapPinListing_Details}>
```

Comment:
What do you think about trying to get rid of some of the divs?  I think we can just add css to replace them.  It would help get rid of some of the extra dom nodes, especially when we have 30+ ( or even 100) listings.

Reply:
If you'd like to work with me on eliminating divs I'm happy to work with you on that.

Reply:
It would be preferable as we'd like to keep the number down.